React/Views/RCTScrollView.m

```diff
@@ -317,8 +317,8 @@ - (void)setFrame:(CGRect)frame
   CGSize boundsSize = self.bounds.size;
 
   self.contentOffset = CGPointMake(
-    MAX(0, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
-    MAX(0, MIN(originalOffset.y, fullContentSize.height - boundsSize.height)));
+    MAX(-contentInset.left, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
```

Comment:
@janicduplessis Yeah, you are absolutely right. We also have pretty similar bug internally. 😢 And I don't have good idea how to fix that. As I wrote internally:

> I understand why it is happening but I have no idea how to fix it properly (without reverting the diff). And the problem seems that there is no enough content size in the rendering moment to scroll to desired offset.
> Conceptually, the problem is that we should not have `scrollOffset` property in the first place, because it is momentary value, not something permanent, and because of this the semantic of this property is unclear.
> 
> So, let's imagine we have `scrollOffset={10,10}` enforced in `render` method.
> Questions:
> When exactly ScrollView must scroll to this offset?
> What if there is no enough content for this?
> Should user actions (touch gestures) reset this value?
> Why should we scroll to this offset only when it is changing?
>
> So, again, because it is momentary, it the perfect world, we have to use `scrollTo` method instead of using this prop.


Any ideas how we can address this?